src/features/views/repos/ViewDataRepo.ts

```diff
@@ -174,6 +174,17 @@ export default class ViewDataRepo {
     this._store.dispatch(columnUpdated([viewId, column]));
   }
 
+  async updateColumnOrder(
+    orgId: number,
+    viewId: number,
+    columnOrder: number[]
+  ) {
+    await this._apiClient.patch<{ order: number[] }>(
+      `/api/orgs/${orgId}/people/views/${viewId}/column_order`,
+      { order: columnOrder }
+    );
```

Comment:
This makes the correct API request, but it does not in any way affect the local store. I reckon the state in the local store needs to either be modified to reflect the new state, or marked as stale so that it's reloaded, right? Otherwise I think this would explain the quirky behavior.

Reply:
Actually no, I don't use the local store. The reason for this is that I figured syncing two different persistent states would make things just that much more difficult. Instead I use the local state for the temporary order of columns. But yes, it appears that I failed to take new columns into account, I need to check for new columns and integrate the new column into the columnOrder.